framework/update/internal/platform/mac/macupdateinstaller.cpp

```
@@ -0,0 +1,256 @@
+bool MacUpdateInstaller::isInPlaceUpdateSupported() const
+{
+    const QString bundlePath = currentBundlePath().toQString();
+    if (bundlePath.isEmpty() || !bundlePath.endsWith(".app")) {
+        return false;
+    }
+
+    if (!QFileInfo::exists(helperPath().toQString())) {
+        return false;
+    }
+
+    // In-place replacement only works if we can write the bundle without
```

Cppcheck (2.21.0)

[style] The function 'isValid' is never used.

(unusedFunction)